language-support/scala/bindings-akka/src/main/scala/com/digitalasset/ledger/client/binding/retrying/CommandRetryFlow.scala

```diff
@@ -70,25 +70,39 @@ object CommandRetryFlow {
             {
               case Ctx(
                     RetryInfo(request, nrOfRetries, firstSubmissionTime, _),
-                    Completion(_, Some(status: Status), _),
+                    result,
                     _,
                   ) =>
-                if (status.code == Code.OK_VALUE) {
-                  PROPAGATE_PORT
-                } else if (
-                  (firstSubmissionTime plus maxRetryTime) isBefore timeProvider.getCurrentTime
-                ) {
-                  RetryLogger.logStopRetrying(request, status, nrOfRetries, firstSubmissionTime)
-                  PROPAGATE_PORT
-                } else if (RETRYABLE_ERROR_CODES.contains(status.code)) {
-                  RetryLogger.logNonFatal(request, status, nrOfRetries)
-                  RETRY_PORT
-                } else {
-                  RetryLogger.logFatal(request, status, nrOfRetries)
-                  PROPAGATE_PORT
+                result match {
+                  case Left(completionFailure) =>
+                    completionFailure match {
+                      case CompletionResponse.NotOkResponse(_, status)
```

Comment:
Why not:

```scala
             case Ctx(
                    RetryInfo(request, nrOfRetries, firstSubmissionTime, _),
                    Left(CompletionResponse.NoOkResponse(_, status)),
                    _,
                  ) =>
```

Instead of 3 levels of `match` unpacking?

Reply:
Changed to a single level of matches but I'm in doubts as to if it makes it easier to read

Reply:
FWIW, I prefer it. I have to keep less state in my head when reading.